### PR TITLE
Mobile debts model

### PR DESCRIPTION
**Pull request by Tonksthebear**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
This adds a model to the debt endpoint in order to ensure we get alerted when there are schema changes

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/8148

## Testing done
Existing tests cover the changes

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by aherzberg — COMMENTED** [review]
There's enough going on here and the debt specs are minimal enough that i think it'd be worth adding an explicit check of the response before and after these changes in the spec to make sure nothing has changed. Unless you'd done that manually already in your own env

**Comment by Tonksthebear** [comment]
@aherzberg I probably went overboard in the quantity of tests I did, but I took the results from the serializer before the changes, hardcoded them into the test, and then changed the serializer again to use the new model. Is there a cleaner way to test the results or is this what you were thinking?

**Comment by aherzberg** [comment]
> @aherzberg I probably went overboard in the quantity of tests I did, but I took the results from the serializer before the changes, hardcoded them into the test, and then changed the serializer again to use the new model. Is there a cleaner way to test the results or is this what you were thinking?

that looks good! About as clean as we can get I think.

**Review by aherzberg — DISMISSED** [review]
LGTM

**Comment by Tonksthebear** [comment]
@aherzberg Sorry needed to fix the spec

**Review by aherzberg — APPROVED** [review]
LGTM

File: modules/mobile/app/serializers/mobile/v0/debts_serializer.rb [hunk 1]
```diff
@@ -36,34 +37,28 @@ def dependent_debts?(debts)
...
+        debt_history = Array.wrap(debt['debtHistory']).map do |history|
+          {
+            date: history['date'],
+            letter_code: history['letterCode'],
+            description: history['description']
+          }
+        end
+
+        Debt.new(id: id || debt['id'],
+                 file_number: debt['fileNumber'],
```
Comment (LindseySaari) [inline modules/mobile/app/serializers/mobile/v0/debts_serializer.rb:49]: Is it ok if any of these debt attributes are nil? Do you need a default value?

Comment (Tonksthebear) [reply modules/mobile/app/serializers/mobile/v0/debts_serializer.rb:49]: Yes nil is ok for the values

**Comment by rmtolmach** [comment]
You had a known, unrelated test failure that was fixed this morning, so I updated your branch.

**Review by rmtolmach — APPROVED** [review]

**Comment by rmtolmach** [comment]
The Require backend review group approval is failing. It shouldn't be! Tag me here or better yet, comment in [the support thread](https://dsva.slack.com/archives/CBU0KDSB1/p1727455387395899) that was opened for this ticket to get us to merge it whenever you're ready.

**Pull request merged**